src/lib/SexyAppBase.cpp

```diff
@@ -704,9 +704,10 @@ bool SexyAppBase::RegistryReadKey(const std::string& theValueName, uint32_t* the
         if (s == "")
             return false;
 
-        uchar* c_str = (uchar*) (SexyStringToString(s)).c_str();
+        std::string ss = SexyStringToString(s);
+        uchar* c_str = (uchar*) ss.c_str();
         if (*theType == REG_SZ) {
-            memcpy(theValue, c_str, 1023);
+            memcpy(theValue, c_str, ss.length() > 1023 ? 1023 : ss.length()+1);
```

Comment:
Is the +1 correct? Suppose `ss.length() == 1023` then the `memcpy` will copy 1024 bytes, right?

It will still be ok, because the caller happens to have a 1024 byte buffer.

BTW, the code is ugly as hell. It is a miracle that it works :-)

Reply:
Sorry, I was trying to be clever. I've pushed a change that removes the ternary. You're right about that +1, it will definitely copy one byte too much in that case.